packages/core/src/libraries/quota.ts

```diff
@@ -0,0 +1,90 @@
+import { EnvSet } from '#src/env-set/index.js';
+import RequestError from '#src/errors/RequestError/index.js';
+import type Queries from '#src/tenants/Queries.js';
+import assertThat from '#src/utils/assert-that.js';
+import { getTenantSubscriptionPlan } from '#src/utils/subscription/index.js';
+import { type FeatureQuota } from '#src/utils/subscription/types.js';
+
+import { type CloudConnectionLibrary } from './cloud-connection.js';
+
+export type QuotaLibrary = ReturnType<typeof createQuotaLibrary>;
+
+export const createQuotaLibrary = (queries: Queries, cloudConnection: CloudConnectionLibrary) => {
+  const {
+    applications: { countNonM2MApplications, countM2MApplications },
+    resources: { findTotalNumberOfResources },
+  } = queries;
+
+  const getTenantUsage = async (key: keyof FeatureQuota): Promise<number> => {
+    if (key === 'applicationsLimit') {
+      return countNonM2MApplications();
+    }
+
+    if (key === 'machineToMachineLimit') {
+      return countM2MApplications();
+    }
+
+    if (key === 'resourcesLimit') {
+      const { count } = await findTotalNumberOfResources();
```

Comment:
Need to filter the default management API

Reply:
updated